src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPASITY = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[CAPASITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size + 1 > elementData.length) {
+            ensureCapacity();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size || checkIndex(index)) { // index >= 0,  index <= size !!!!
```

Comment:
Let's remove all comments
```suggestion
        if (index == size || checkIndex(index)) {
```

Reply:
done